### ISSUE TITLE
[Bug]: result of show create view is not correct after altering the view 

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#7817]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):d54992c96efce308c17d37a638fb147a8fce6863
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

MO:
mysql> SHOW CREATE VIEW v0;
+------+----------------------------------------------------------------------------+
| View | Create View                                                                |
+------+----------------------------------------------------------------------------+
| v0   | CREATE VIEW v0 AS SELECT t00.a, t01.a AS b FROM t00 LEFT JOIN t01 USING(a) |
+------+----------------------------------------------------------------------------+
1 row in set (0.01 sec)

mysql> 
mysql> ALTER VIEW v0 AS SELECT t00.a FROM t00;
Query OK, 0 rows affected (0.07 sec)

mysql> SELECT t00.a FROM t00;
+------+
| a    |
+------+
|    1 |
|    2 |
+------+
2 rows in set (0.07 sec)

mysql> SELECT * FROM v0;
+------+
| a    |
+------+
|    1 |
|    2 |
+------+
2 rows in set (0.01 sec)

mysql> SHOW CREATE VIEW v0;
+------+----------------------------------------+
| View | Create View                            |
+------+----------------------------------------+
| v0   | ALTER VIEW v0 AS SELECT t00.a FROM t00 |
+------+----------------------------------------+
1 row in set (0.15 sec)

MYSQL:
mysql> SHOW CREATE VIEW v0;
+------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
| View | Create View                                                                                                                                                                               | character_set_client | collation_connection |
+------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
| v0   | CREATE ALGORITHM=UNDEFINED DEFINER=`dump`@`localhost` SQL SECURITY DEFINER VIEW `v0` AS select `t00`.`a` AS `a`,`t01`.`a` AS `b` from (`t00` left join `t01` on((`t00`.`a` = `t01`.`a`))) | utf8mb4              | utf8mb4_0900_ai_ci   |
+------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
1 row in set (0.00 sec)

mysql> 
mysql> ALTER VIEW v0 AS SELECT t00.a FROM t00;
Query OK, 0 rows affected (0.00 sec)

mysql> SELECT t00.a FROM t00;
+------+
| a    |
+------+
|    1 |
|    2 |
+------+
2 rows in set (0.00 sec)

mysql> SELECT * FROM v0;
+------+
| a    |
+------+
|    1 |
|    2 |
+------+
2 rows in set (0.00 sec)

mysql> SHOW CREATE VIEW v0;
+------+----------------------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
| View | Create View                                                                                                                | character_set_client | collation_connection |
+------+----------------------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
| v0   | CREATE ALGORITHM=UNDEFINED DEFINER=`dump`@`localhost` SQL SECURITY DEFINER VIEW `v0` AS select `t00`.`a` AS `a` from `t00` | utf8mb4              | utf8mb4_0900_ai_ci   |
+------+----------------------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
1 row in set (0.01 sec)

### Steps to Reproduce

```Markdown
DROP TABLE IF EXISTS t00;
DROP TABLE IF EXISTS t01;
DROP VIEW IF EXISTS v0;
CREATE TABLE t00(a INTEGER);
INSERT INTO t00 VALUES (1),(2);
CREATE TABLE t01(a INTEGER);
INSERT INTO t01 VALUES (1);

CREATE VIEW v0 AS SELECT t00.a, t01.a AS b FROM t00 LEFT JOIN t01 USING(a);
SELECT t00.a, t01.a AS b FROM t00 LEFT JOIN t01 USING(a);
SELECT * FROM v0 WHERE b >= 0;
SHOW CREATE VIEW v0;

ALTER VIEW v0 AS SELECT t00.a FROM t00;
SELECT t00.a FROM t00;
SELECT * FROM v0;
SHOW CREATE VIEW v0;
SELECT t00.a FROM t00;
DROP TABLE IF EXISTS t00;
DROP TABLE IF EXISTS t01;
DROP TABLE IF EXISTS v0;
```

**Comment by iceTTTT** [comment]
Pr #7825 merged

**Comment by aressu1985** [comment]
fixed: 4410d8618193455fc0be25d3af17d19b5bbc1a0e
